### PR TITLE
plots modify: complain if template does not exist

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here. If the CLI API is changed, I have updated [tab completion scripts](https://github.com/iterative/dvc/tree/master/scripts/completion).

* [x] ❌ I will check DeepSource, CodeClimate, and other sanity checks below. (We consider them recommendatory and don't expect everything to be addressed. Please fix things that actually improve code or fix bugs.)

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/plots/__init__.py [hunk 1]
```diff
@@ -15,6 +18,11 @@ def diff(self, *args, **kwargs):
     def modify(self, path, props=None, unset=None):
         from dvc.dvcfile import Dvcfile
 
+        props = props or {}
+        template = props.get("template")
+        if template:
+            _evaluate_templatepath(self.repo, template)
```
Comment (skshetry) [inline dvc/repo/plots/__init__.py:24]:
```suggestion
            self.repo.plot_templates.get_template(template)
```
If #3994 gets merged first

File: dvc/repo/plots/__init__.py [hunk 2]
```diff
@@ -32,4 +40,4 @@ def modify(self, path, props=None, unset=None):
         out.verify_metric()
 
         dvcfile = Dvcfile(self.repo, out.stage.path)
-        dvcfile.dump(out.stage, update_pipeline=True)
+        dvcfile.dump(out.stage, update_pipeline=True, no_lock=True)
```
Comment (skshetry) [inline dvc/repo/plots/__init__.py:43]: `plots modify` should not try to generate lockfile.

File: dvc/repo/plots/__init__.py [hunk 3]
```diff
@@ -1,3 +1,6 @@
+from dvc.repo.plots.show import _evaluate_templatepath
```
Comment (skshetry) [inline dvc/repo/plots/__init__.py:1]: This is fixed by refactoring on #3994

File: tests/func/plots/__init__.py [hunk 4]
```diff
@@ -5,11 +5,11 @@
 
 @pytest.fixture
 def run_copy_metrics(tmp_dir, run_copy):
-    def run(file1, file2, commit=None, tag=None, **kwargs):
+    def run(file1, file2, commit=None, tag=None, single_stage=True, **kwargs):
```
Comment (skshetry) [inline tests/func/plots/__init__.py:8]: Need to fix these ugly hacks: singele_stage being the default

**Comment by efiop** [comment]
@skshetry Could you rebase, please?

File: tests/func/plots/test_plots.py [hunk 5]
```diff
@@ -538,3 +540,48 @@ def test_load_metric_from_dict_yaml(tmp_dir):
...
+        plots_no_cache=["metric.json"],
+        name="copy-metrics",
+        single_stage=False,
+    )
+
+    (tmp_dir / PIPELINE_LOCK).unlink()
+    dvc.plots.modify(
+        "metric.json", props={"template": relpath(custom_template)}
+    )
+    assert not (tmp_dir / PIPELINE_LOCK).exists()
```
Comment (skshetry) [inline tests/func/plots/test_plots.py:580]: Not sure how to write a test that qualifies for "is not modified".

Comment (pared) [reply tests/func/plots/test_plots.py:580]: @skshetry How about checking mtime of lockfile?

Comment (skshetry) [reply tests/func/plots/test_plots.py:580]: Thanks @pared for suggestion, but `mtime` is too far, I think. I'll just leave it as-is for now.

**Pull request merged**